src/helpers/issue.ts

```diff
@@ -7,46 +7,18 @@ import {
   StreamlinedComment,
   UserType,
 } from "../types";
-import { checkRateLimitGit } from "../utils";
 import { LogReturn } from "../adapters/supabase";
 import { Payload, Context } from "../types";
 
 type PromiseType<T> = T extends Promise<infer U> ? U : never;
 
 async function getAllIssueEvents(context: Context) {
-  type Event = PromiseType<ReturnType<typeof context.event.octokit.issues.listEvents>>["data"][0];
+  if (!context.payload.issue) return;
 
-  const payload = context.event.payload as Payload;
-  if (!payload.issue) return;
-
-  let shouldFetch = true;
-  let page = 1;
-
-  const events = [] as Event[];
-
-  while (shouldFetch) {
-    // Fetch issue events
-
-    const repo = payload.repository.name;
-    const owner = payload.repository.owner.login;
-
-    const response = await context.event.octokit.issues.listEvents({
-      owner: owner,
-      repo: repo,
-      issue_number: payload.issue.number,
-      per_page: 100,
-      page: page,
-    });
-
-    await checkRateLimitGit(response?.headers);
-
-    if (response.data.length > 0) {
-      events.push(...(response.data as Event[]));
-      page++;
-    } else {
-      shouldFetch = false;
-    }
-  }
```

Comment:
It doesn't look like you replaced the functionality of looping through every page of results.